### PR TITLE
Ftr: support get/post/put/patch in rest protocol

### PR DESCRIPTION
**Pull request by benbochong**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: protocol/rest/server/server_impl/go_restful_server.go [hunk 1]
```diff
@@ -82,23 +83,39 @@ func (grs *GoRestfulServer) Start(url common.URL) {
...
 	}
 	ws.Path(restMethodConfig.Path).
 		Produces(strings.Split(restMethodConfig.Produces, ",")...).
 		Consumes(strings.Split(restMethodConfig.Consumes, ",")...).
 		Route(ws.Method(restMethodConfig.MethodType).To(rf))
-	grs.container.Add(ws)
+	grs.container.Add(ws)*/
 
+	ws := &restful.WebService{}
+	ws.Path(restServiceConfig.Path)
```
Comment (Patrick0308) [inline protocol/rest/server/server_impl/go_restful_server.go:98]: It will make user can't define same service path.

Comment (benbochong) [reply protocol/rest/server/server_impl/go_restful_server.go:98]: Config different root path for different service

File: protocol/rest/config/reader/rest_config_reader.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ func initMethodConfigMap(rc *config.RestServiceConfig, consumes string, produces
 	mcm := make(map[string]*config.RestMethodConfig, len(rc.RestMethodConfigs))
 	for _, mc := range rc.RestMethodConfigs {
 		mc.InterfaceName = rc.InterfaceName
-		mc.Path = rc.Path + mc.Path
+		//mc.Path = rc.Path + mc.Path
```
Comment (AlexStocks) [inline protocol/rest/config/reader/rest_config_reader.go:93]: delete it if u think it is of no usage.

File: protocol/rest/rest_protocol.go [hunk 3]
```diff
@@ -75,9 +75,10 @@ func (rp *RestProtocol) Export(invoker protocol.Invoker) protocol.Exporter {
 	}
 	rp.SetExporterMap(serviceKey, exporter)
 	restServer := rp.getServer(url, restServiceConfig.Server)
-	for _, methodConfig := range restServiceConfig.RestMethodConfigsMap {
+	/*for _, methodConfig := range restServiceConfig.RestMethodConfigsMap {
 		restServer.Deploy(methodConfig, server.GetRouteFunc(invoker, methodConfig))
-	}
+	}*/
```
Comment (AlexStocks) [inline protocol/rest/rest_protocol.go:80]: the same as above.

File: protocol/rest/server/server_impl/go_restful_server.go [hunk 4]
```diff
@@ -82,23 +83,39 @@ func (grs *GoRestfulServer) Start(url common.URL) {
 
 // Publish a http api in go-restful server
 // The routeFunc should be invoked when the server receive a request
-func (grs *GoRestfulServer) Deploy(restMethodConfig *config.RestMethodConfig, routeFunc func(request server.RestServerRequest, response server.RestServerResponse)) {
-	ws := &restful.WebService{}
+func (grs *GoRestfulServer) Deploy(restServiceConfig *config.RestServiceConfig, invoker protocol.Invoker) {
+	/*ws := &restful.WebService{}
```
Comment (zouyx) [inline protocol/rest/server/server_impl/go_restful_server.go]: Delete it if useless

File: protocol/rest/server/server_impl/go_restful_server.go (hunk 1 above)
Comment (Patrick0308) [reply protocol/rest/server/server_impl/go_restful_server.go]: At first, we can't ask user must set server path config. And we should let user more flexible to config path, like spring mvc.

Comment (benbochong) [reply protocol/rest/server/server_impl/go_restful_server.go]: You don't have to set the root path. The following configurations are completely equivalent.

```
services:
  "CourseProvider":
    registry: "etcd"
    protocol : "rest"
    interface : "com.llll.Course"
    loadbalance: "random"
    warmup: "100"
    cluster: "failover"
    rest_path: "/courses"
    rest_produces: "application/json;charset=utf-8"
    rest_consumes: "application/json;charset=utf-8,*/*"
    methods:
    - name: "ViewList"
      rest_path: "/"
      rest_method: "GET"
    - name: "ViewOne"
      rest_path: "/{id}"
      rest_path_params: "0:id"
      rest_method: "GET"
    - name: "CreateDoc"
      rest_path: "/"
      rest_body: 0
      rest_method: "POST"
    - name: "EditDoc"
      rest_path: "/{id}"
      rest_path_params: "0:id"
      rest_body: 1
      rest_method: "POST"
```

```
services:
  "CourseProvider":
    registry: "etcd"
    protocol : "rest"
    interface : "com.llll.Course"
    loadbalance: "random"
    warmup: "100"
    cluster: "failover"
    rest_produces: "application/json;charset=utf-8"
    rest_consumes: "application/json;charset=utf-8,*/*"
    methods:
    - name: "ViewList"
      rest_path: "/courses"
      rest_method: "GET"
    - name: "ViewOne"
      rest_path: "/courses/{id}"
      rest_path_params: "0:id"
      rest_method: "GET"
    - name: "CreateDoc"
      rest_path: "/courses"
      rest_body: 0
      rest_method: "POST"
    - name: "EditDoc"
      rest_path: "/courses/{id}"
      rest_path_params: "0:id"
      rest_body: 1
      rest_method: "POST"
```

Comment (benbochong) [reply protocol/rest/server/server_impl/go_restful_server.go]: Restful protocol specifies architecture constraints and principles. Usually, a service corresponds to a resource，the rootpath is the resource name.

**Comment by Patrick0308** [comment]
I think #676 is better, so I will close this pr.

**Pull request closed without merging**